src/store.c

```diff
@@ -79,31 +85,59 @@ struct rill_store
 // -----------------------------------------------------------------------------
 
 static struct encoder store_encoder(
-        struct rill_store *store, struct indexer *indexer)
+        struct rill_store *store,
+        struct indexer *indexer,
+        struct vals* vals,
+        uint64_t offset)
 {
     return make_encoder(
-            store->vma + store->head->data_off,
-            store->vma + store->vma_len,
-            store->vals, indexer);
-}
-
-static struct decoder store_decoder(struct rill_store *store)
-{
-    return make_decoder(
-            store->vma + store->head->data_off,
+            store->vma + offset,
             store->vma + store->vma_len,
-            store->vals, store->index);
+            vals,
+            indexer);
 }
 
 static struct decoder store_decoder_at(
-        struct rill_store *store, size_t key_idx, uint64_t off)
+        struct rill_store *store,
+        size_t key_idx,
+        uint64_t curr_off,
+        enum rill_col column)
 {
+    assert(column == rill_col_a || column == rill_col_b);
+
+    size_t offset = 0;
+    size_t offset_end = 0;
+    struct index* lookup = NULL;
+    struct index* index = NULL;
+
+    if (column == rill_col_a) {
```

Comment:
Note that instead of the asserts, you may consider using a switch statement since we have a bunch of warnings that can tell you whether you're missing one of the enums in your switch case.

Reply:
TIL, thanks! Will fix.